### PR TITLE
Prometheus

### PR DESCRIPTION
**Pull request by max-jacobs**
### :link: Related Issues

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,9 @@
 ### Changed
 * Updated `hotels-oss-parent` to 5.0.0 (was 4.0.1).
 
+### Added
+* Support for prometheus metrics.
```
Comment (massdosage) [inline CHANGELOG.md]:
```suggestion
* Support for Prometheus metrics.
```

File: waggle-dance-core/pom.xml [hunk 2]
```diff
@@ -97,6 +97,11 @@
       <artifactId>micrometer-registry-graphite</artifactId>
     </dependency>
 
+    <dependency>
```
Comment (massdosage) [inline waggle-dance-core/pom.xml]: To be really pedantic this should go below the jmx one if we want to keep these alpha-sorted by groupid.artifactid where possible.

File: waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/metrics/MonitoringConfigurationTestContext.java [hunk 3]
```diff
@@ -27,4 +28,10 @@ GraphiteConfiguration graphiteConfiguration() {
     return conf;
   }
 
+  @Bean
+  PrometheusConfiguration prometheusConfiguration() {
```
Comment (massdosage) [inline waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/metrics/MonitoringConfigurationTestContext.java]: So, just to confirm, we now *always* expose a Prometheus endpoint? i.e. it's not something you can turn on or off?

Comment (max-jacobs) [reply waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/metrics/MonitoringConfigurationTestContext.java]: No I don't think so. The new Prometheus dependency will automatically create the PrometheusRegistry bean only. To expose the endpoint, when the jar is started up with some configuration, that configuration will need to include which endpoints to expose - this addition will happen here I think https://github.com/ExpediaGroup/apiary-federation/tree/master/templates

...subject to testing of course

Comment (massdosage) [reply waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/metrics/MonitoringConfigurationTestContext.java]: OK, make sense, we should document this then as per my other comment.

File: README.md [hunk 4]
```diff
@@ -310,6 +310,10 @@ The following snippet shows a typical Graphite configuration:
 | `graphite.poll-time`                                  | No       | Amount of time between Graphite polls. Defaults to `5000`. |
 | `graphite.poll-time-unit`                             | No       | Time unit of `graphite.poll-time` - this is [the list of allowed values](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/TimeUnit.html). Defaults to `MILLISECONDS`. |
 
+Prometheus can also be used to gather metrics. This can be done by enabling the Prometheus endpoint in the configuration:
```
Comment (patduin) [inline README.md]: can you add there what the default "url" would be will just make it a little easier to find I assume it's normally printed startup logs?

Comment (max-jacobs) [reply README.md]: I think the default url is just `/prometheus`

Comment (max-jacobs) [reply README.md]: Done

Comment (patduin) [reply README.md]: ah ok that trivial, excellent :)

**Pull request merged**